internal/app/networkd/pkg/networkd/networkd.go

```diff
@@ -280,9 +280,17 @@ func (n *Networkd) decideHostname() (hostname string, domainname string, address
 	address = net.ParseIP("127.0.1.1")
 	hostname = fmt.Sprintf("%s-%s", "talos", strings.ReplaceAll(address.String(), ".", "-"))
 
+	// Sort interface names alphabetically so we can ensure parsing order
+	interfaceNames := make([]string, len(n.Interfaces))
+	for intName := range n.Interfaces {
+		interfaceNames = append(interfaceNames, intName)
+	}
+
```

Comment:
I guess `sort.Strings(interfaceNames)` is missing here, as otherwise it's equivalent to the code it was before

Reply:
yep, thanks. just realized myself. give it a refresh.